### PR TITLE
Implement unit wrapper interface and basic Worker Wrapper

### PR DESCRIPTION
**Pull request by 0x416C657852**

### CODE REVIEW

File: src/UnitWrapper/WorkerWrapper.h [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            {
+                if (unit->getDistance(i) < smallestDistance)
+                {
+                    smallestDistance = unit->getDistance(i);
+                    closest = i;
+                }
+            }
+            if (closest)
+            {
+                unit->gather(closest);
```
Comment (krogenth) [inline src/UnitWrapper/WorkerWrapper.h:48]:
```suggestion
                unit->gather(closest);
                currJob = Jobs::MineMinerals;
```

File: src/UnitWrapper/WorkerWrapper.h [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+class WorkerWrapper : public UnitWrapper
+{
+protected:
+    Jobs CurrJob = Jobs::None;
+
+public:
+    WorkerWrapper(BWAPI::Unit u) : UnitWrapper(u){};
+    bool isBusy()
+    {
+        if (CurrJob == Jobs::None)
```
Comment (krogenth) [inline src/UnitWrapper/WorkerWrapper.h:22]: I'm ok leaving this as-is for now, but we'll need to revisit the `isBusy` method to be a bit more granular in what it defines as "busy." Things like checking if the worker is carrying gas/minerals, if it currently is morphing/going to morph, etc.

File: src/UnitWrapper/WorkerWrapper.h [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        {
+            return true;
+        }
+        else
+        {
+            return false;
+        }
+    }
+
+    void OnFrame()
```
Comment (krogenth) [inline src/UnitWrapper/WorkerWrapper.h]:
```suggestion
    void onFrame()
```

File: src/UnitWrapper/UnitWrapper.h [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    UnitWrapper(BWAPI::Unit u)
+    {
+        unit = u;
+        unitID = u->getID();
+    };
+
+    virtual bool isBusy() = 0;
+    virtual void OnFrame() = 0;
+
+    void Draw()
```
Comment (krogenth) [inline src/UnitWrapper/UnitWrapper.h:21]: We should try to come up with a reasonable universal naming convention for this method. I think `displayInfo` from #21 is a potentially reasonable choice.

I also think this method should be overloaded/virtual, to allow displaying the unit's current state(i.e. - a worker's Job) below the unit(`drawTextMap()`).

**Review by krogenth — DISMISSED** [review]

**Review by Soulsteeler123 — DISMISSED** [review]

**Review by krogenth — APPROVED** [review]

**Review by Soulsteeler123 — APPROVED** [review]

**Pull request merged**